### PR TITLE
Prevent non string taxonomy names generating warnings or errors.

### PR DESCRIPTION
**Pull request by costdev**
Trac ticket: https://core.trac.wordpress.org/ticket/56338

### CODE REVIEW

File: tests/phpunit/tests/taxonomy.php [hunk 1]
```diff
@@ -125,7 +125,12 @@ public function test_taxonomy_exists_unknown() {
 		$this->assertFalse( taxonomy_exists( rand_str() ) );
 		$this->assertFalse( taxonomy_exists( '' ) );
 		$this->assertFalse( taxonomy_exists( 0 ) );
-		$this->assertFalse( taxonomy_exists( null ) );
+		$this->assertFalse( taxonomy_exists( 1 ), 'Did not return false with (int) 1' );
+		$this->assertFalse( taxonomy_exists( 0.0 ), 'Did not return false with (float) 0.0' );
+		$this->assertFalse( taxonomy_exists( 1.1 ), 'Did not return false with (float) 1.1' );
+		$this->assertFalse( taxonomy_exists( null ), 'Did not return false with null' );
+		$this->assertFalse( taxonomy_exists( array(), 'Did not return false with an array' ) );
+		$this->assertFalse( taxonomy_exists( new stdClass(), 'Did not return false with an object' ) );
```
Comment (costdev) [inline tests/phpunit/tests/taxonomy.php]: These tests should ideally be in a data provider. Given the existing tests aren't, this seems like a matter for another ticket.

**Comment by mukeshpanchal27** [comment]
@costdev Can you please add test coverage for the special characters?

**Comment by costdev** [comment]
> @costdev Can you please add test coverage for the special characters?

Hi @mukeshpanchal27, I'm not sure what you mean. Which special characters will trigger an illegal offset warning/error?

**Review by mukeshpanchal27 — APPROVED** [review]
After a quick discussion with @costdev, we don't need to check the special characters test here.

LGTM 🚀

File: tests/phpunit/tests/taxonomy.php [hunk 2]
```diff
@@ -125,7 +125,12 @@ public function test_taxonomy_exists_unknown() {
 		$this->assertFalse( taxonomy_exists( rand_str() ) );
 		$this->assertFalse( taxonomy_exists( '' ) );
 		$this->assertFalse( taxonomy_exists( 0 ) );
-		$this->assertFalse( taxonomy_exists( null ) );
+		$this->assertFalse( taxonomy_exists( 1 ), 'Did not return false with (int) 1' );
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/taxonomy.php]:
```suggestion
		$this->assertFalse( taxonomy_exists( 1 ), 'Did not return false with (int) 1' );
	}

	public function test_taxonomy_exists_invalid_parameter_type()
		$this->assertFalse( taxonomy_exists( 1 ), 'Did not return false with (int) 1' );
```

Nitpick but they seem subtly different tests.

File: tests/phpunit/tests/taxonomy.php [hunk 3]
```diff
@@ -125,7 +125,41 @@ public function test_taxonomy_exists_unknown() {
...
+	 * @return array
+	 */
+	public function data_taxonomy_exists_should_return_false_with_non_string_taxonomy() {
+		return array(
+			'array'        => array( array() ),
+			'object'       => array( new stdClass() ),
+			'bool (true)'  => array( true ),
+			'bool (false)' => array( false ),
+			'null'         => array( null ),
+			'integer (0)'  => array( 0 ),
```
Comment (costdev) [inline tests/phpunit/tests/taxonomy.php:158]: `(int) 0` is already tested in the `test_taxonomy_exists_unknown()` test method, but is included here for completeness and context.

**Review by peterwilsoncc — APPROVED** [review]

**Comment by audrasjb** [comment]
committed in https://core.trac.wordpress.org/changeset/53869

**Pull request closed without merging**